code/Spikes/SW_UART/SW_UART.ino

```diff
@@ -0,0 +1,65 @@
+/**
+ * @brief Interrupt based UART Example
+ * @author F.Bartsch
+ */
+
+// Include SoftwareSerial library
+#include <SoftwareSerial.h>
+
+// Create SoftwareSerial object
+SoftwareSerial mySerial(10, 11); // RX, TX
+
+// set up glabal data 
+String rxString = "";
+boolean rxStringCompleteFlag = false;
+
+void setup()
+{
+  // Set data rate for HW Serial port on Arduino board Pin 0 and Pin 1
+  Serial.begin(9600);
+
+  while (!Serial)
+  {
+    ; // wait for serial port to connect.
+  }
+
+  rxString.reserve(200);
+
+  // Set data rate for the SW SoftwareSerial port
+  mySerial.begin(9600);
+}
+
+void loop() {
+  // print the string when a newline arrives:
+  if (rxStringCompleteFlag) 
+  {
+    mySerial.write("STRING RECEIVED");
+    
+    // clear the string:
+    rxString = "";
+    rxStringCompleteFlag = false;
+  }
+}
+
+/**
+ * @brief ISR of UART RX
+ */
+void serialEvent() {
+  
+  while (Serial.available()) 
+  {
+    // catch the incoming char
+    char inChar = (char)Serial.read();
+    
+    // add it to the inputString:
+    rxString += inChar;
```

Comment:
Ja das geht natuerlich auch. Ringbuffer scheint mir sicherer, da nicht irgendjemand einfach das x in einem commit aendern kann. 
Fuer die erste Implementierung wuerde ich auch nur das x nehmen. Ist einfach und schnell gemacht :).